client/src/app/(root)/layout.tsx

```diff
@@ -1,21 +1,36 @@
 import { PropsWithChildren } from "react";
 
+import { dehydrate, Hydrate, QueryClient } from "@tanstack/react-query";
 import type { Metadata } from "next";
 
+import { client } from "@/lib/queryClient";
+import { queryKeys } from "@/lib/queryKeys";
+
 import Sidebar from "@/containers/sidebar";
 
 export const metadata: Metadata = {
   title: "Explore | 4Growth",
   description: "Explore | 4Growth",
 };
 
-export default function ExploreLayout({ children }: PropsWithChildren) {
+export default async function ExploreLayout({ children }: PropsWithChildren) {
+  const queryClient = new QueryClient();
+  await queryClient.prefetchQuery({
+    queryKey: queryKeys.sections.all.queryKey,
+    queryFn: async () =>
+      client.sections.searchSections.query({
+        fetchOptions: { cache: "no-cache" },
```

Comment:
It seems that ts-rest has an opinionated way of instantiating the query client, prob worth looking:
https://ts-rest.com/docs/react-query/ssr#react-server-components